### PR TITLE
Add makeIndependent with requires_grad argument

### PR DESCRIPTION
**Pull request by junjihashimoto**
Add a requires_grad argument to make a tensor of requires_grad=False, because running_mean argument of batchNorm function requires a tensor with requires_grad = False.

### CODE REVIEW

**Review by tscholak — APPROVED** [review]
LGTM.
This seems to be a band-aid to a more fundamental problem with the semantics around `makeIndependent` and how it is implemented in the API. Let's consider alternatives later.

**Comment by junjihashimoto** [comment]
Thx!
Pytorch module has parameter of requires_grad=true and buffer of requires_grad=false,
so It may be good to add a buffer type for untyped api.
https://pytorch.org/docs/stable/generated/torch.nn.Module.html
